pluggy.py

```diff
@@ -412,14 +423,16 @@ def _verify_hook(self, hook, hookimpl):
                 "Plugin %r\nhook %r\nhistoric incompatible to hookwrapper" %
                 (hookimpl.plugin_name, hook.name))
 
-        for arg in hookimpl.argnames:
-            if arg not in hook.argnames:
-                raise PluginValidationError(
-                    "Plugin %r\nhook %r\nargument %r not available\n"
-                    "plugin definition: %s\n"
-                    "available hookargs: %s" %
-                    (hookimpl.plugin_name, hook.name, arg,
-                     _formatdef(hookimpl.function), ", ".join(hook.argnames)))
+        # positional arg checking
+        notinspec = set(hookimpl.argnames) - set(hook.argnames)
+        if notinspec:
+            raise PluginValidationError(
+                "Plugin %r for hook %r\nhookimpl definition: %s\n"
```

Comment:
Isn't #43 about supporting *default* values? 

Let's see if we are using the same nomenclature first. Considering this function:

```python
def foobar(x, y, z=4):
    pass  
```

* positional arguments: caller passes by position so order matters: `foobar(1, 2, 10)`
* keyword arguments: caller passes using keyword syntax so order does not matter: `foobar(y=2, x=1, z=10)`
* default arguments: caller does not need to pass them on: `foobar(1, 2)  # z = 4`

As far as I understand, hooks are always called using *keyword arguments* so order is not an issue, that's why my comment was referring to "Positional arguments" as confusing.

Note that I'm talking about the *caller* here, not the definition. The definition might for example force the caller to call a function using only keywords:

```python
def foobar(*, x, y, z=4):
    pass
```

(Python 3 only of course)

AFAIU, `pluggy` always calls using keyword arguments, so specs are implicitly defined as:

```python
def pytest_deselectitems(*, items, session):
    pass
```

And that issue is completely separated from *default* values.

Am I missing something?